### PR TITLE
Cairo code for spy_message_to_l1

### PR DESCRIPTION
**Pull request by Arcticae**
**Stack**:
- #2284
- #2283
- #2281
- #2280 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: snforge_std/src/cheatcodes/message_to_l1.cairo [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+        MessagesToL1 { messages }
+    }
+}
+
+trait MessageToL1FilterTrait {
+    /// Filter messages emitted by a sender of a given [`ContractAddress`]
+    fn sent_by(self: @MessagesToL1, contract_address: ContractAddress) -> MessagesToL1;
+    /// Filter messages emitted by a receiver of a given ethereum address (`felt252`)
+    fn sent_to(self: @MessagesToL1, to_address: EthAddress) -> MessagesToL1;
```
Comment (Draggu) [inline snforge_std/src/cheatcodes/message_to_l1.cairo:57]: Shouldn't this be `received_by`?

Comment (Arcticae) [reply snforge_std/src/cheatcodes/message_to_l1.cairo:57]: It implies that the message has been received when in fact it hasn't really

**Review by Draggu — APPROVED** [review]

File: snforge_std/src/cheatcodes/message_to_l1.cairo [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        };
+    }
+}
+
+fn is_sent(
+    messages: @MessagesToL1, expected_sent_by: @ContractAddress, expected_message: @MessageToL1,
+) -> bool {
+    let mut i = 0;
+    let mut is_emitted = false;
+    while i < messages.messages.len() {
```
Comment (ksew1) [inline snforge_std/src/cheatcodes/message_to_l1.cairo:133]: `messages.messages` might be confusing. Maybe we should name it differently?

**Review by tomek0123456789 — APPROVED** [review]

**Comment by Arcticae** [comment]
✓ Commit merged in pull request [#2284](https://github.com/foundry-rs/starknet-foundry/pull/2284)

**Pull request closed without merging**